2.2_comb_logic.ipynb

```diff
@@ -342,30 +342,30 @@
     "    val pe1_data   = Output(UInt(16.W))\n",
     "  })\n",
     "\n",
-    "  ???  \n",
+    "  ???\n",
     "}\n",
-    "class ArbiterTester(c: Arbiter) extends PeekPokeTester(c) {\n",
+    "\n",
+    "test(new Arbiter) { c =>\n",
     "  import scala.util.Random\n",
     "  val data = Random.nextInt(65536)\n",
-    "  poke(c.io.fifo_data, data)\n",
+    "  c.io.fifo_data.poke(data.U)\n",
     "  \n",
     "  for (i <- 0 until 8) {\n",
-    "    poke(c.io.fifo_valid, (i>>0)%2)\n",
-    "    poke(c.io.pe0_ready,  (i>>1)%2)\n",
-    "    poke(c.io.pe1_ready,  (i>>2)%2)\n",
+    "    c.io.fifo_valid.poke((((i >> 0) % 2) != 0).B)\n",
```

Comment:
Does it make sense to use Decoupled operations like `enqueue` and `expectDequeue` here? Perhaps this example should go into a further tutorial module on test drivers / composition, but I would consider those functions "proper" usage and should be the first example test with a ready-valid interface.

Reply:
I think this makes sense but I don't have the time to get this written, I've put an hour into it and I'm still debugging. There are good examples of enqueue and dequeue in 2.6, I will make a comment point to it for now. If you have time it would be great if you can provide the proper code here.

Reply:
Yeah, I think the right solution is to use another example where one wouldn't naturally use a Decoupled, and save the first Decoupled example to use the enqueue/dequeue ops. But I think in the short term a adding some text in the tutorial here is fine, along the lines of:

>this example is being refactored, but the typical recommended way in ChiselTest to interface with Decoupled ready/valid interfaces is with higher-level functions `.enqueue` and `.expectDequeue`

I just don't want anyone walking out of this thinking low-level pokes are the right style.

Side: I'm reluctant to refactor this right now because there's a bit more boilerplate about initializing the enqueue and dequeue interfaces, so the intro really doesn't belong here.